include/pybind11/common.h

```diff
@@ -153,8 +153,10 @@
 #define PYBIND11_SLICE_OBJECT PyObject
 #define PYBIND11_FROM_STRING PyUnicode_FromString
 #define PYBIND11_STR_TYPE ::pybind11::str
+#define PYBIND11_NONZERO "__bool__"
```

Comment:
Could we rename this to `BOOL_ATTR` or something along those lines to stick to the python-3-derived name here rather than python-2-derived, like we do with the above (e.g. "BYTES").